src/main/java/com/corundumstudio/socketio/misc/CompositeIterator.java

```diff
@@ -39,7 +39,7 @@ public boolean hasNext() {
             }
             return false;
         }
-        return currentIterator.hasNext();
+        return true; // can only be reached when currentIterator.hasNext() is true
```

Comment:
can you explain more here?

Reply:
@mrniko
```
    public boolean hasNext() {
        if (currentIterator == null || !currentIterator.hasNext()) {
            while (listIterator.hasNext()) {
                Iterator<T> iterator = listIterator.next();
                if (iterator.hasNext()) {
                    currentIterator = iterator;
                    return true;
                }
            }
            return false;
        } 
```
The line below can only be reached when `currentIterator != null && currentIterator.hasNext() == true` 
(equivalent of `!(currentIterator == null || !currentIterator.hasNext())` which is basically `else` clause)  
earlier this line returned currentIterator.hasNext() which is always true due to the previous statement
```
        return true; // can only be reached when currentIterator.hasNext() is true
    }
```
(Even static code analyzer shows that it will always be true)

Reply:
@mrniko Will you merge?)